pkg/plugins/mellanox/mellanox_plugin.go

```diff
@@ -76,32 +76,41 @@ func (p *MellanoxPlugin) OnNodeStateChange(old, new *sriovnetworkv1.SriovNetwork
 		if !isMlnxNicAndInNode(iface.PciAddress, new) {
 			continue
 		}
-		fwCurrent, fwNext, err := getMlnxNicFwData(iface.PciAddress)
+		fwData, err := getMlnxNicFwData(iface.PciAddress)
 		if err != nil {
 			return false, false, err
 		}
-		if fwCurrent.enableSriov != fwNext.enableSriov || fwCurrent.totalVfs != fwNext.totalVfs ||
-			fwCurrent.linkType != fwNext.linkType {
-			needReboot = true
-			break
-		}
-		attrs := &mlnxNic{totalVfs: -1, firstPort: fwCurrent.firstPort}
+		attrs := &mlnxNic{totalVfs: -1, firstPort: fwData.firstPort}
 		requireChange := false
 
-		if fwCurrent.totalVfs != iface.NumVfs {
+		if fwData.totalVfs != iface.NumVfs {
 			attrs.totalVfs = iface.NumVfs
 			requireChange = true
 		}
-		if iface.NumVfs == 0 && fwCurrent.enableSriov {
+		if iface.NumVfs == 0 && fwData.enableSriov {
 			attrs.enableSriov = false
 			requireChange = true
-		} else if iface.NumVfs > 0 && !fwCurrent.enableSriov {
+		} else if iface.NumVfs > 0 && !fwData.enableSriov {
 			attrs.enableSriov = true
 			requireChange = true
 		}
-		if fwCurrent.linkType != EthLinkType && fwCurrent.linkType != PreconfiguredLinkType && fwCurrent.linkType != UknownLinkType {
-			attrs.linkType = EthLinkType
-			requireChange = true
+		if iface.LinkType != "" {
+			if iface.LinkType != EthLinkType && iface.LinkType != InfinibandLinkType {
```

Comment:
Do you need to create this PR on top of the dual port fix you have?

Reply:
Yes, It can be done after one of these 2 get merged since there will be a conflict between them

Reply:
also it will be the same case for the switchdev mode PR